### PR TITLE
[onert] Add helper functions for TrainableBackendContext

### PR DESCRIPTION
**Pull request by ragmani**
This commit adds helper functions for TrainableBackendContext.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: runtime/onert/backend/train/BackendContext.cc [hunk 1]
```diff
@@ -30,9 +30,9 @@ namespace train
 
 ITensorRegistry *BackendContext::genTensors() { return basic::genTensors(*this); }
 
-FunctionMap BackendContext::genKernels()
+backend::FunctionMap BackendContext::genKernels()
 {
-  FunctionMap ret;
+  backend::FunctionMap ret;
```
Comment (ragmani) [inline runtime/onert/backend/train/BackendContext.cc:35]: There was a conflict with the namespace train in the core, so I added `backend::`.

File: runtime/onert/core/include/backend/basic/BackendContextHelpers.h [hunk 2]
```diff
@@ -24,6 +24,7 @@
 #include "util/logging.h"
 #include "backend/ITensorRegistry.h"
 #include "backend/BackendContext.h"
+#include "backend/train/TrainableBackendContext.h"
```
Comment (glistening) [inline runtime/onert/core/include/backend/basic/BackendContextHelpers.h:27]: @ragmani May I ask why you don't put these change under `backend/train/BackendContextHelpers.h`? You've put the changes under `train/` so far.

Comment (ragmani) [reply runtime/onert/core/include/backend/basic/BackendContextHelpers.h:27]: OK I will introduce a new helper header. There will still be modifications of the `basic/BackendContextHelpers.h` to reduce code duplication.

**Comment by ragmani** [comment]
PTAL @glistening

**Comment by ragmani** [comment]
PTAL @glistening

**Comment by glistening** [comment]
@ragmani As I said offline yesterday, I don't fully understand why `train` is under `basic`. I need more time to understand why.

**Review by hseok-oh — APPROVED** [review]

**Review by glistening — APPROVED** [review]

**Pull request merged**